File: MAP/LAB7-MAP/LAB7-MAP/ControllerFile/Controller.cs
using MAP_Lab7.Statements;
using MAP_Lab7.Repository;
using MAP_Lab7.Model;
using System;

namespace MAP_Lab7.ControllerFile
{
    public class Controller
    {
        private IPrgStateRepo repo;

        public Controller(IPrgStateRepo r)
        {
            repo = r;
        }
        
        public void executeOneStep()
        {
            PrgState ps = repo.GetCurrentProgram();
            IExeStack<Statement> ex = ps.ExeStack;
            if (!ex.IsEmpty())
            {
                Statement stmt = ex.PopS();
                stmt.Execute(ps);
            }
            Console.WriteLine("----------------------------------------------\n\n");
        }

        public void executeAll()
        {
            PrgState ps = repo.GetCurrentProgram();
            IExeStack<Statement> es = ps.ExeStack;
            while (!es.IsEmpty())
            {
                executeOneStep();
                repo.logPrgStateExec();
            }
            Console.WriteLine("----------------------------------------------\n\n");
        }
    }
}

File: MAP/Lab3/src/Controller/Controller.java
package Controller;
import Repository.*;
import Model.*;
import Model.Statement.*;

public class Controller
{
    private IPrgStateRepo ipsr;
    public Controller(IPrgStateRepo ip)
    {
        ipsr = ip;
    }
    public void executeOneStep()
    {
        PrgState ps = ipsr.getCurrentProgram();
        IExecStack<Statement> ex = ps.getExecStack();
        if(! ex.isEmpty())
        {
            Statement instr = ex.pop();
            instr.execute(ps);
            System.out.println(ps);

        }
        System.out.println("----------------------------------------------");
    }
    public void executeAll()
    {
        PrgState ps = ipsr.getCurrentProgram();
        IExecStack<Statement> es = ps.getExecStack();

        ipsr.logDateProgStateExec();
        while(! es.isEmpty())
        {
            executeOneStep();
            ipsr.logPrgStateExec();
        }
        System.out.println("----------------------------------------------");
    }

    public void add(PrgState prgState){
        ipsr.addPrgState(prgState);
    }
}